src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,41 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.service.exceptions.PasswordException;
+import core.basesyntax.service.exceptions.UserRegistrationException;
 
 public class RegistrationServiceImpl implements RegistrationService {
+
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        userValidate(user);
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserRegistrationException("This user with login already exists");
+        }
```

Comment:
This is also a part of validation process. It's better to move it to "userValidate" method
```suggestion
```